proxy.go

```diff
@@ -33,26 +35,22 @@ var errProxyTimeout = errors.New("all peers timed out")
 //     case the code just waits for one to finish. If the total 'proxy_timeout'
 //     is hit at any point, the method returns immediately with an error and
 //     cancels any running requests.
-func (vs *version) proxy(w http.ResponseWriter, r *http.Request, peers []string) ([]byte, error) {
+func (vs *version) proxy(peers []string) (*http.Response, string, error) {
 	responses := make(chan proxyResponse, len(peers))
-	cancel := make(chan struct{})
-	defer close(cancel)
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(vs.sequins.config.Sharding.ProxyTimeout.Duration))
```

Comment:
Would that let me cancel all this stuff if the original client went away? That does seem useful, but it means lots more threading through.